### PR TITLE
feature (core): support fog for none logarithmic depth buffer 

### PR DESCRIPTION
**Pull request by gchoqueux**
add support fog for none logarithmic depth buffer

### CODE REVIEW

File: src/Core/TileMesh.js [hunk 1]
```diff
@@ -85,7 +85,9 @@ TileMesh.prototype.changeState = function changeState(state) {
 };
 
 TileMesh.prototype.setFog = function setFog(fog) {
-    this.material.setFogDistance(fog);
+    if (fog != undefined) {
```
Comment (peppsac) [inline src/Core/TileMesh.js]: In which situation would `setFog(...)` would be called with `fog = undefined` (ie: are you sure the bug isn't elsewhere?)

**Review by peppsac — CHANGES_REQUESTED** [review]
Wording of your git commits do not follow #518

File: src/Renderer/Shader/TileFS.glsl [hunk 2]
```diff
@@ -78,7 +78,8 @@ void main() {
             float depth = gl_FragDepthEXT / gl_FragCoord.w;
             float fogIntensity = 1.0/(exp(depth/distanceFog));
         #else
-            float fogIntensity = 1.0;
+            float depth = gl_FragCoord.z / gl_FragCoord.w;
+            float fogIntensity = 1.0/(exp(depth/distanceFog));
```
Comment (autra) [inline src/Renderer/Shader/TileFS.glsl]: this line could be moved out of the #if/#else

**Review by autra — DISMISSED** [review]
LGTM, just one nits if you care to correct.

For general information, I tested with Epiphany on Ubuntu 16.04: it doesn't support log depth buffer.

**Comment by gchoqueux** [comment]
There are several browsers don't support depth buffer

File: src/Core/TileMesh.js (hunk 1 above)
Comment (gchoqueux) [reply src/Core/TileMesh.js]: Because the fogDistance isn't always defined, we can give it a default value. 
it's reason break on planar View

Comment (peppsac) [reply src/Core/TileMesh.js]: Could you instead fix the caller to not call `setFog` with an undefined value please?

Comment (gchoqueux) [reply src/Core/TileMesh.js]: done

**Review by peppsac — APPROVED** [review]
Thanks.

**Pull request merged**